### DISCUSSION TITLE
I want to create additional &quot;author&quot; sub-origins in the cascade

### DISCUSSION THREAD

**Discussion by aarongustafson** [WebWeWant/webwewant.fyi#95] [Wants]
The first step of the cascade is to sort rules into layers of &quot;origin &amp; importance&quot;—where rules in each layer override rules in the previous layer, regardless of specificity:

- user agent normal
- user normal
- author normal
- (animations)
- author `!important`
- user `!important`
- user agent `!important`

We can use `!important` to create a new author layer with extra-high specificity, but it&#39;s not generally a good idea and will override animations. The `:where()` selector will allow zero-specificity selectors, which is also a step in the right direction, but I think there are many use-cases for defining &quot;origin&quot; layers like &quot;resets&quot; / &quot;third-party tools&quot; / &quot;brand defaults&quot; / &quot;theme&quot; / &quot;patterns&quot; / &quot;components&quot; - so that component styles override pattern styles and so on: a cascade of author origins.

The cascade is a powerful concept, and the foundation of CSS, but authors only have  direct control over specificity and source order. This leads to specificity wars between different layers of the system that might share specificity, but have a clear division in terms of &quot;origin&quot; type.

Cross-posted to the CSS Working Group: [https://github.com/w3c/csswg-drafts/issues/4470](https://github.com/w3c/csswg-drafts/issues/4470)

<hr>

https://webwewant.fyi/wants/82/

**Comment by Trapfether** [comment]
Hello, I think you might be interested to hear about CSS Layers. It is currently experimental, but it would enable what I believe you are asking for, among other use-cases.

[https://developer.mozilla.org/en-US/docs/Web/CSS/@layer](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer)
